src/main/java/com/teragrep/pth_06/planner/BloomFilterTempTable.java

```diff
@@ -0,0 +1,219 @@
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+
+package com.teragrep.pth_06.planner;
+
+import com.teragrep.blf_01.Token;
+import org.apache.spark.util.sketch.BloomFilter;
+import org.jooq.*;
+import org.jooq.impl.DSL;
+import org.jooq.types.ULong;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import static com.teragrep.pth_06.jooq.generated.bloomdb.Bloomdb.BLOOMDB;
+import static org.jooq.impl.SQLDataType.*;
+
+/**
+ * SQL Temp table filled with bloom filters of the parent table filter types, search term token set is
+ * inserted into the temp table filter columns. Each row of the parent table can be compared against this temp table.
+ * <p>
+ * Parent Table Schema:
+ * <ol>
+ *     <li>id PK</li>
+ *     <li>partition_id FK journaldb.logfile.id</li>
+ *     <li>filter_type_id FK bloomdb.filtertype.id</li>
+ *     <li>filter - bloomfilter byte array</li>
+ * </ol>
+ * <p>
+ * Temp Table Schema:
+ * <ol>
+ *     <li>id PK</li>
+ *     <li>termId - count of the current search term</li>
+ *     <li>typeId FK bloomdb.filtertype.id</li>
+ *     <li>filter - bloomfilter bytes with only the search term token set inserted</li>
+ * </ol>
+ * Parent table create table example:
+ * <p>
+ * <code>
+ *     CREATE TABLE `example` ( `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT, `partition_id` bigint(20) unsigned NOT NULL, `filter_type_id` bigint(20) unsigned NOT NULL, `filter` longblob NOT NULL, PRIMARY KEY (`id`), UNIQUE KEY `partition_id` (`partition_id`), KEY `example_ibfk_1` (`filter_type_id`), CONSTRAINT `example_ibfk_1` FOREIGN KEY (`filter_type_id`) REFERENCES `filtertype` (`id`) ON DELETE CASCADE, CONSTRAINT `example_ibfk_2` FOREIGN KEY (`partition_id`) REFERENCES `journaldb`.`logfile` (`id`) ON DELETE CASCADE ) ENGINE=InnoDB AUTO_INCREMENT=54787 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci
+ * </code>
+ * @see com.teragrep.pth_06.jooq.generated.bloomdb.Bloomdb
+ */
+public final class BloomFilterTempTable {
+    private static final Logger LOGGER = LoggerFactory.getLogger(BloomFilterTempTable.class);
+
+    private final DSLContext ctx;
+    private final Table<?> parentTable;
+    private final Table<Record> tableName;
+    private final long bloomTermId;
+    private final Set<Token> tokenSet;
+    private final List<Condition> cache;
```

Comment:
why a cache is being used?

Reply:
to limit the SQL operations if the Condition is already generated. Should I refactor them to work without caches?